src/main/java/org/broadinstitute/hellbender/engine/ReadInput.java

```diff
@@ -0,0 +1,170 @@
+package org.broadinstitute.hellbender.engine;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.broadinstitute.hellbender.utils.Utils;
+
+import java.io.File;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Class to represent a Read-containing input file. Tools should declare @Argument-annotated fields of
+ * this type (or Collections of this type), and the engine will automatically discover
+ * them at runtime.
```

Comment:
This is not true -- we don't auto-discover `ReadInputs` in the same way that we auto-discover `FeatureInputs`